static/app/views/explore/multiQueryMode/locationUtils.tsx

```diff
@@ -198,6 +204,34 @@ export function useDeleteQueryAtIndex() {
   );
 }
 
+export function getSamplesTargetAtIndex(
+  index: number,
+  queries: ReadableExploreQueryParts[],
+  row: Record<string, any>,
+  location: Location
+): Location {
+  const queryToUpdate = queries[index];
+  if (!queryToUpdate) {
+    return location;
+  }
+
+  const queryString = queryToUpdate.query ?? '';
+  const search = new MutableSearch(queryString);
+  for (const groupBy of queryToUpdate.groupBys) {
+    const value = row[groupBy];
+    search.setFilterValues(groupBy, [value]);
+  }
```

Comment:
minor nit, feels slightly more readable to me:
```suggestion
  queryToUpdate.groupBys.forEach(groupBy => {
    search.setFilterValues(groupBy, [row[groupBy]]);
  });
```